crypto/polyval-generic.c

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * POLYVAL: hash function for GCM-SIV.
+ * Code based on crypto/ghash-generic.c
```

Comment:
Can you add a comment that briefly explains how this implementation works?  Especially how it differs from ghash-generic.